jdaviz/configs/cubeviz/plugins/parsers.py

```diff
@@ -140,13 +140,20 @@ def _return_spectrum_with_correct_units(flux, wcs, metadata, data_type, target_w
             category=UserWarning)
         sc = Spectrum1D(flux=flux, wcs=wcs)
 
-    # TODO: Can the unit be defined in a different keyword, e.g., CUNIT1?
     if target_wave_unit is None and hdulist is not None:
-        cunit_key = 'CUNIT3'
+        found_target = False
         for ext in ('SCI', 'FLUX', 'PRIMARY'):  # In priority order
-            if ext in hdulist and cunit_key in hdulist[ext].header:
-                target_wave_unit = u.Unit(hdulist[ext].header[cunit_key])
+            if found_target:
                 break
+            if ext not in hdulist:
+                continue
+            hdr = hdulist[ext].header
+            # The WCS could be swapped or unswapped.
+            for cunit_key in ('CUNIT3', 'CUNIT1'):
```

Comment:
I think this broke a test, I'll need to investigate.

```
    def test_fits_image_hdu_with_microns(image_cube_hdu_obj_microns, cubeviz_helper):
        # Passing in data_label keyword as posarg.
        cubeviz_helper.load_data(image_cube_hdu_obj_microns, 'has_microns')
    
        assert len(cubeviz_helper.app.data_collection) == 3
        assert cubeviz_helper.app.data_collection[0].label == 'has_microns[FLUX]'
    
        flux_cube = cubeviz_helper.app.data_collection[0].get_object(Spectrum1D, statistic=None)
>       assert flux_cube.spectral_axis.unit == u.um
E       assert Unit("m") == Unit("um")
```

Reply:
that looks like a `wcs.set`-produced error?  I think wcslib's WCS always forces wavelength->m

Reply:
This function was put in to circumvent that... lol sigh

Reply:
Ah, yep.  Then it's probably a logical error related to `found_target`.  But it looks sound to me so I'm no use.

Reply:
Must be something real subtle. No worries, I plan to investigate when I get the chance. Thanks for your feedback!